### ISSUE TITLE
P4_14 parser exceptions are not recognized by p4c and p4test

### ISSUE THREAD

**Issue by againwq** [p4lang/p4c#4198]
I tried to use p4test to translate my P4_14 code into P4_16 code, but it threw the following error:
**includes/parser.p4(291):syntax error, unexpected IDENTIFIER, expecting ;
 default : parse_error p4_pe_default
                       ^^^^^^^^^^^^^
includes/parser.p4(306):syntax error, unexpected ;
    parser_drop;
               ^
[--Werror=overlimit] error: 8 errors encountered, aborting compilation**

my P4_14 code is:
``` P4
parser parse_signature_info{
    extract(signature_info);
    return select(current(0,8)) {
	NDNTYPE_SGV : parse_signature_value;
	default : parse_error p4_pe_default;
    }
}

parser_exception p4_pe_default {
    parser_drop;
}
```
How can I solve this problem?

**Comment by fruffy** [comment]
What is the command you used? The default compilation standard is now P4-16. So you might have to use the `--std p4-14` flag.

**Comment by againwq** [comment]
My command is 
```
p4c --target bmv2 --arch v1model --std p4-14 my_switch.p4
```
and
```
p4test --std p4-14 my_switch.p4 --pp my_switch16.p4

``` 
my p4c version is 1.2.4.2, enviroment is ubuntu20.04. I install p4c by "sudo apt install p4lang-p4c"

**Comment by againwq** [comment]
How can I solve this problem? Or is there any other way to compile P4_14 code with parser expection, or to use other p4c-supported syntax to implement packet drop in the parser stage in the p4_14

**Comment by vgurevich** [comment]
@againwq in P4_16 the model is quite different. Moreover, it is architecture-dependent.  Here are the basics:

1. P4_16 defines two final parser states: `accept` and `reject`
2. There are 3 way to transition to the `reject` state:
    a. Exceptional situations (like running our of packet data during parsing) cause the parser to implicitly transition to the `reject` state.
   b. Another option to transition to the `reject` state is via the `verify()` statement
   c. The third option is explicit transition
3. Handling of the packets that reach the `reject` state is defined by the architecture, not by P4 language. Some architectures might, for example, drop the packet, while others might not drop it, etc. 
   * If you want to drop a packet, you need to consult the reference for the specific architecture you are using

Basically, you need to rewrite that portion of your P4_14 program manually.
